### ISSUE TITLE
[Overview page] Filter namespaces by label

### ISSUE THREAD

**Issue by lucasponce** [kiali/kiali#1522]
There is this scenario where cluster admins label each namespace with the team name. This way allows admins determine really quickly who has to ping to in case of troubles in that namespace.

The feature would be adding one more filter option into Overview page for filtering ns by label(s). It could be one label or more than one.

JIRA:
https://issues.jboss.org/browse/KIALI-2996

**Comment by lucasponce** [comment]
I even think that this feature can be extended to all list pages too.
I think labels are already mapped and brought to the UI, and adding a similar a text filter option based on labels can be easy to implement.

**Comment by aljesusg** [comment]
We require some changes in backend here...

Options

1. New endpoint to check labels for namespaces
2. Add a queryparam in namespaces endpoint to filter by labels

I prefer the 2º option

**Comment by jotak** [comment]
Question: currently in the merged PR, if I understand correctly, on the overview page the labels that are looked for are the ones on namespaces, is it correct?

It's positive because now I can see, for instance, the "istio-enabled=true" label that I've set on them, it's a valuable information that totally makes sense to show.

However, for filtering when coming from other list pages, it is a bit weird because the filters refer to different objects, it's like comparing apples and oranges. For instance, if from the application page I filter with "version=v1" and see 3 matching applications, then go to overview page: the filter is kept but here it shows 0 namespace / applications because the filter is unrelated.

I'm raising this issue but to be honest I'm not sure how it should be addressed. What perhaps I would expect here is to see all the overview cards, with health items filtered to show only those for version=v1, as it was intended when I filtered from apps page. But it clearly contradicts with the goal to have namespace-based label filters. So there's a dissonance.

Perhaps a solution in-between could be just to clear the active label filters while switching from *list pages to overview, because they don't relate to the same thing ?
